### PR TITLE
Automation of CEPH-9230:snap&clone of an imported image

### PR DESCRIPTION
**Pull request by VasishtaShastry**
Automated CEPH-9230 - RBD-import/export:Import an rbd image and create snaps and clones
and addted it to new suite

new file:   suites/nautilus/rbd/tier-2_rbd_regression.yaml
new file:   tests/rbd/rbd_snap_clone_imported_image.py
new file:   suites/pacific/rbd/tier-2_rbd_regression.yaml
new file:   tests/rbd/exceptions.py
modified:   tests/rbd/rbd_utils.py

Updates:
Feb 7 - Addressed optimisation suggestions to python f strings
        and exception handling using userdefined exceptions

Signed-off-by: Vasishta <vashastr@redhat.com>

# Description

Please include Automation development guidelines. Source of Test case - New Feature/Regression Test/Close loop of customer BZs
<details>

<summary>click to expand checklist</summary>

- [ ] Create a test case in Polarion reviewed and approved.
- [ ] Create a design/automation approach doc. Optional for tests with similar tests already automated.
- [ ] Review the automation design
- [ ] Implement the test script and perform test runs
- [ ] Submit PR for code review and approve
- [ ] Update Polarin Test with Automation script details and update automation fields
- [ ] If automation is part of Close loop, update BZ flag qe-test_coverage “+” and link Polarion test
</details>

### CODE REVIEW

**Comment by VasishtaShastry** [comment]
TBD - 
1) add 5.x suite for same test
2) Update polarion ID

**Comment by VasishtaShastry** [comment]
Had a discussion with Pragadeeswaran and got some inputs
1) Suite has some unnecessary copy-pasted configs, needs to be removed
2) Suite name needs to be changed to provide summary if the tests
3) new Module needs more documentation on the purpose of the module
4) vertical spacing in suite needs to be standardised
5) in suite, EOL needs to have just one new line character

**Comment by VasishtaShastry** [comment]
@rakeshgm @Gopi-Patta @sunilkumarn417 PR is ready for review

**Comment by VasishtaShastry** [comment]
> @rakeshgm @Gopi-Patta @sunilkumarn417 PR is ready for review
A gentle reminder

File: tests/rbd/rbd_snap_clone_imported_image.py
Comment (sunilkumarn417) [inline tests/rbd/rbd_snap_clone_imported_image.py:8]: please update the doc strings as per this ref doc, https://www.sphinx-doc.org/en/master/usage/extensions/example_google.html#example-google

File: tests/rbd/rbd_snap_clone_imported_image.py
Comment (sunilkumarn417) [inline tests/rbd/rbd_snap_clone_imported_image.py:24]: as per the test case configuration, no parameters are passed except test metadata. here we are using same to initialize Rbd class. is this needed?

File: tests/rbd/rbd_snap_clone_imported_image.py
Comment (sunilkumarn417) [inline tests/rbd/rbd_snap_clone_imported_image.py:42]: would it be a good idea to add library module for Rbd to handle all supported operations available in `Rbd()`.

File: tests/rbd/rbd_snap_clone_imported_image.py
Comment (VasishtaShastry) [reply tests/rbd/rbd_snap_clone_imported_image.py]: Done

File: tests/rbd/rbd_snap_clone_imported_image.py
Comment (VasishtaShastry) [reply tests/rbd/rbd_snap_clone_imported_image.py]: It was observed that its was required (example for ceph_nodes)

**Comment by VasishtaShastry** [comment]
Recent log link - http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-KXP6Q9
cc @sunilkumarn417 @rakeshgm

File: tests/rbd/rbd_utils.py
Comment (rakeshgm) [inline tests/rbd/rbd_utils.py:180]: Handle using exceptions.

File: tests/rbd/rbd_utils.py
Comment (rakeshgm) [inline tests/rbd/rbd_utils.py:134]: you could also refactor like this. 

```
cmd = f"dd if=/dev/urandom of={filename} bs=4 count=5M"
if not self.exec_cmd(
    cmd=cmd,
    long_running=True,
):
    raise CreateFileImportError("file import failed")
```            
it could be similar for the following functions too.

Comment (VasishtaShastry) [reply tests/rbd/rbd_utils.py:134]: Sure
Is it for 1. some optimization best practice or 2. a code standard that we want to follow in this repo ?

Comment (rakeshgm) [reply tests/rbd/rbd_utils.py:134]: I don't think this is a coding standard, I think this is a small optimization. The only standard I see in my comment is making use of `format strings`.

**Comment by VasishtaShastry** [comment]
Positive scenario - http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-MRLQ3K

**Review by sunilkumarn417 — APPROVED** [review]
LGTM. 
@VasishtaShastry , we could merge this PR once @rakeshgm approves.

Added DNM label to skip mergify, we could remove once we have approval from Rakesh.

**Pull request merged**
